### ISSUE TITLE
[Slider] Accessibility content description for virtual view subtree 

### ISSUE THREAD

**Issue by jaredrummler** [material-components/material-components-android#2176]
When Explore by Touch is enabled (Google TalkBack), the slider announces the value and not the label associated with the value, causing confusion to users who rely on accessibility features.

**Describe the solution you'd like**
The `BaseSlider.AccessibilityHelper` is a private class and, as far as I can tell, the content description for the virtual views can not be overridden. It would be useful to set the content description for each slider step/value. Currently, TalkBack announces the following text when focused on a virtual view on the slider:

> "0%, Slider, Use volume keys to adjust"

It would be more useful to replace the percentage with the associated label for each value:

> "[Label at 0%], Slider, Use volume keys to adjust"

Adding an `AccessibilityDelegate` is not an option as the `Slider` view has its own private implementation which is not accessible to consumers of the library. It would be helpful to make this public so a custom implementation of an `AccessibilityDelegate` would be possible or adding a method to set the content description for each virtual view.

**Comment by ymarian** [comment]
Hi I think makes sense, thanks for the suggestion
